packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -442,4 +444,31 @@ export class ProjectModel {
             await func();
         });
     }
+
+    async getWarehouseFromCache(
+        projectUuid: string,
+    ): Promise<WarehouseCatalog> {
+        const warehouses = await this.database(CachedWarehouseTableName)
+            .select(['warehouse'])
+            .where('project_uuid', projectUuid)
+            .limit(1);
+        if (warehouses.length > 0) return warehouses[0].warehouse;
+        return {};
```

Comment:
Prefer to return undefined / not found error - so we can distinguish a cache-miss from cache-hit with an empty warehouse schema